repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManager.java

```diff
@@ -144,14 +146,22 @@ private void setImpersonator() {
     private void setTenant() {
         filterDirector.requestHeaderManager().putHeader(OpenStackServiceHeader.TENANT_NAME.toString(), cachableToken.getTenantName());
         if(!this.isDelagable && !this.isTenanted) {
-            filterDirector.requestHeaderManager().putHeader(OpenStackServiceHeader.TENANT_ID.toString(), cachableToken.getTenantId());
+            if (sendTenantIdQuality) {
```

Comment:
Busted. I started to write tests then wept at the current state of testing around this class.

I'll go add these now.